### PR TITLE
Field distortion and transverse diffusion map

### PR DESCRIPTION
**Pull request by ftoschi**
**What is the problem / what does the code in this PR do**

This PR includes the possibility to include field distortion coming from a initial-to-final position map (typically coming from COMSOL simulations). This is possible by setting the `field_distortion_model` to `comsol`, while by default it is set to `none`: in case this is specified, the map stored in `field_distortion_comsol_map` will be used to map the radial initial position of a given event to a final distorted position. 

In addition, this PR enables the possibility of using transverse diffusion map(s), assuming that these are divided into radial and azimuthal components as it would come from a 2D-axisymmetric simulation. 

**WFSim needs you:**
 - Please add a test for this PR, as a bare minimum, make sure it's covered in coveralls!
   - _Will provide_
 - Can you add a docsting to all your functions?

*Pay attention:*
 - Due to databases being needed for testing, making a PR from your own fork will typically NOT run the tests. If you then merge master might break

### CODE REVIEW

**Review by zhut19 — APPROVED** [review]

File: wfsim/core/s2.py [hunk 1]
```diff
@@ -355,17 +376,34 @@ def s2_pattern_map_diffuse(n_electron, z, xy, config, resource):
...
+            diffusion_constant_transverse = getattr(config, 'diffusion_constant_transverse', 0)
+            diffusion_constant_radial = diffusion_constant_transverse
+            diffusion_constant_azimuthal = diffusion_constant_transverse
 
-        drift_time_mean = - z / drift_velocity_liquid + drift_time_gate  # Add gate time for consistancy?
-        hdiff_stdev = np.sqrt(2 * diffusion_constant_transverse * drift_time_mean)
-
-        hdiff = np.random.normal(0, 1, (np.sum(n_electron), 2)) * \
-            np.repeat(hdiff_stdev, n_electron, axis=0).reshape((-1, 1))
+        drift_time_mean = - z / drift_velocity_liquid
```
[the remark this answers is not in the document]
Comment (zhut19) [reply wfsim/core/s2.py:397]: Continue in #248
